modules/internal/ChapelArray.chpl

```diff
@@ -610,6 +616,46 @@ module ChapelArray {
     return chpl__distributed(d, chpl__buildDomainExpr((...ranges)));
   }
 
+  //
+  // Array-view utility functions
+  //
+  proc chpl__isArrayView(arr) param {
+    use Reflection;
+    param isSlice = if canResolveMethod(arr, "isSliceArrayView") then arr.isSliceArrayView() else false;
+    param isRankChange = if canResolveMethod(arr, "isRankChangeArrayView") then arr.isRankChangeArrayView() else false;
+    param isReindex = if canResolveMethod(arr, "isReindexArrayView") then arr.isReindexArrayView() else false;
```

Comment:
Couldn't these be written like this:
``` chapel
  proc isSliceArrayView( x ) param return false;
  proc isSliceArrayView ( x: ArrayViewSliceArr ) param return true;
  param isSlice = isSliceArrayView(arr);
```

etc?

It seems unnecessary to use canResolve here.